diff --git a/bits/simd_x86.h b/bits/simd_x86.h
index f00ad4f..11b5297 100644
--- a/bits/simd_x86.h
+++ b/bits/simd_x86.h
@@ -2405,11 +2405,11 @@ namespace std::__detail
         _S_sqrt(_TV __x)
         {
           if constexpr (__vec_builtin_sizeof<_TV, 2, 64> and _Flags._M_have_avx512fp16())
-            return __builtin_ia32_sqrtph512_mask_round(__x, _TV(), -1, _X86Round::_CurDirection);
+            return __builtin_ia32_sqrtph512_mask_round(__x, _TV(), -1, static_cast<int>(_X86Round::_CurDirection));
           else if constexpr (__vec_builtin_sizeof<_TV, 4, 64>)
-            return __builtin_ia32_sqrtps512_mask(__x, _TV(), -1, _X86Round::_CurDirection);
+            return __builtin_ia32_sqrtps512_mask(__x, _TV(), -1, static_cast<int>(_X86Round::_CurDirection));
           else if constexpr (__vec_builtin_sizeof<_TV, 8, 64>)
-            return __builtin_ia32_sqrtpd512_mask(__x, _TV(), -1, _X86Round::_CurDirection);
+            return __builtin_ia32_sqrtpd512_mask(__x, _TV(), -1, static_cast<int>(_X86Round::_CurDirection));
           else if constexpr (__vec_builtin_sizeof<_TV, 2, 32> and _Flags._M_have_avx512fp16())
             return __builtin_ia32_sqrtph256_mask(__x, _TV(), -1);
           else if constexpr (__vec_builtin_sizeof<_TV, 4, 32>)
@@ -2421,7 +2421,7 @@ namespace std::__detail
           else if constexpr (__vec_builtin_sizeof<_TV, 4>)
             return __builtin_ia32_sqrtps(__x);
           else if constexpr (__vec_builtin_sizeof<_TV, 8>)
-            return __builtin_ia32_sqrtps(__x);
+            return __builtin_ia32_sqrtpd(__x);
           else
             static_assert(false);
         }
