diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c371f6..e79be70 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -352,12 +352,23 @@ target_include_directories(hpipm
 
 target_link_libraries(hpipm blasfeo)
 
-install(TARGETS hpipm EXPORT hpipmConfig
+install(TARGETS hpipm EXPORT hpipmTargets
 	LIBRARY DESTINATION lib
 	ARCHIVE DESTINATION lib
 	RUNTIME DESTINATION bin)
 
-install(EXPORT hpipmConfig DESTINATION cmake)
+if (HPIPM_FIND_BLASFEO)
+	include(CMakePackageConfigHelpers)
+	configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Config.cmake.in
+		"${CMAKE_CURRENT_BINARY_DIR}/hpipmConfig.cmake"
+		INSTALL_DESTINATION lib/cmake/hpipm)
+	install(FILES
+		"${CMAKE_CURRENT_BINARY_DIR}/hpipmConfig.cmake"
+		DESTINATION lib/cmake/hpipm)
+	install(EXPORT hpipmTargets
+		FILE hpipmTargets.cmake
+		DESTINATION lib/cmake/hpipm)
+endif()
 
 file(GLOB_RECURSE HPIPM_HEADERS "include/*.h")
 install(FILES ${HPIPM_HEADERS} DESTINATION ${HPIPM_HEADERS_INSTALLATION_DIRECTORY})
diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
new file mode 100644
index 0000000..6b92dd1
--- /dev/null
+++ b/cmake/Config.cmake.in
@@ -0,0 +1,7 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+find_dependency(blasfeo)
+include("${CMAKE_CURRENT_LIST_DIR}/hpipmTargets.cmake")
+
+check_required_components(hpipm)
